### ISSUE TITLE
Creating (or splitting) a cell scrolls the notebook up

### ISSUE THREAD

**Issue by mattharrison** [jupyterlab/jupyterlab#14728]
## Description

I have a (long?) notebook (it is a chapter from a book) with multiple cells. 

Often when a create a new cell or split a cell, Jupyter will scroll. It usually scroll up
to near the top of the document.

## Reproduce

1. Grab a large notebook
2. add a cell in the middle (or split a cell)
3. Jupyter might scroll moving the selected cell and cursor off the screen

## Expected behavior

When you split a cell, there should be no scrolling.

When you create a new cell, that cell should be on screen.

## Context

- Operating System and version: Mac 12.5.1
- Browser and version: Chrome 114.0.5735.106
- JupyterLab version: 4.0.2

<details><summary>Troubleshoot Output</summary>
<pre>

$PATH:
	/Users/matt/.envs/menv/bin
	/opt/homebrew/bin
	/opt/homebrew/sbin
	/usr/local/bin
	/usr/bin
	/bin
	/usr/sbin
	/sbin
	/Library/TeX/texbin

sys.path:
	/Users/matt/.envs/menv/bin
	/opt/homebrew/Cellar/python@3.10/3.10.12/Frameworks/Python.framework/Versions/3.10/lib/python310.zip
	/opt/homebrew/Cellar/python@3.10/3.10.12/Frameworks/Python.framework/Versions/3.10/lib/python3.10
	/opt/homebrew/Cellar/python@3.10/3.10.12/Frameworks/Python.framework/Versions/3.10/lib/python3.10/lib-dynload
	/Users/matt/.envs/menv/lib/python3.10/site-packages
	/Users/matt/Dropbox/work/js/jupyterlab-scroll
	/Users/matt/Dropbox/work/pylib/rubber3
	/Users/matt/Dropbox/work/pylib/rst2odp
	/Users/matt/Dropbox/work/pylib/MealMaker
	/Users/matt/Dropbox/work/pylib/countdown
	/Users/matt/Dropbox/work/pylib/svgslicer
	/Users/matt/Dropbox/work/pylib/jupyter_manip
	/opt/homebrew/opt/python-tk@3.10/libexec

sys.executable:
	/Users/matt/.envs/menv/bin/python3.10

sys.version:
	3.10.12 (main, Jun  7 2023, 00:37:42) [Clang 14.0.0 (clang-1400.0.29.202)]

platform.platform():
	macOS-12.5.1-arm64-arm-64bit

which -a jupyter:
	/Users/matt/.envs/menv/bin/jupyter

pip list:
	Package                    Version                         Editable project location
	-------------------------- ------------------------------- ---------------------------------------------
	aiofiles                   22.1.0
	aiohttp                    3.8.4
	aiosignal                  1.3.1
	aiosqlite                  0.18.0
	alembic                    1.9.1
	altgraph                   0.17.3
	anyio                      3.6.2
	appdirs                    1.4.4
	appnope                    0.1.3
	argon2-cffi                21.3.0
	argon2-cffi-bindings       21.2.0
	arrow                      1.2.3
	asn1crypto                 1.5.1
	asttokens                  2.2.1
	async-lru                  1.0.3
	async-timeout              4.0.2
	attrs                      22.2.0
	Babel                      2.11.0
	backcall                   0.2.0
	beautifulsoup4             4.11.1
	big                        0.7.1
	black                      23.1.0
	bleach                     5.0.1
	blinker                    1.4
	blis                       0.7.9
	bokeh                      3.1.0
	boto3                      1.26.91
	botocore                   1.29.91
	Brotli                     1.0.9
	build                      0.10.0
	bump2version               1.0.1
	cachetools                 5.3.0
	catalogue                  2.0.8
	catboost                   1.2
	cattrs                     22.2.0
	certifi                    2022.12.7
	cffi                       1.15.1
	cfgv                       3.3.1
	charset-normalizer         2.1.1
	click                      8.1.3
	cloudpickle                2.0.0
	colorama                   0.4.6
	colour                     0.1.5
	comm                       0.1.2
	confection                 0.0.4
	contourpy                  1.0.6
	copier                     7.2.0
	countdown                  0.1                             /Users/matt/Dropbox/work/pylib/countdown
	coverage                   7.1.0
	cryptography               39.0.0
	cycler                     0.11.0
	cymem                      2.0.7
	Cython                     0.29.35
	dash-core-components       2.0.0
	dash-html-components       2.0.0
	dash-table                 5.0.0
	databricks-cli             0.17.4
	dataclasses-json           0.5.7
	db-dtypes                  1.0.5
	debugpy                    1.6.4
	decorator                  5.1.1
	defusedxml                 0.7.1
	distlib                    0.3.6
	docker                     6.0.1
	docutils                   0.15.2
	dtreeviz                   2.1.3
	dunamai                    1.17.0
	editables                  0.3
	en-core-web-sm             3.5.0
	entrypoints                0.4
	et-xmlfile                 1.1.0
	exceptiongroup             1.1.0
	executing                  1.2.0
	faiss-cpu                  1.7.4
	fastjsonschema             2.16.2
	fbs                        1.1.9
	feature-engine             1.5.2
	feedgenerator              1.9
	filelock                   3.9.0
	flake8                     6.0.0
	flake8-markdown            0.4.0
	Flask                      2.2.2
	fonttools                  4.38.0
	fqdn                       1.5.1
	frozendict                 2.3.4
	frozenlist                 1.3.3
	fsspec                     2023.3.0
	fst-pso                    1.8.1
	funcy                      2.0
	future                     0.18.2
	FuzzyTM                    2.0.5
	gensim                     4.3.0
	gitdb                      4.0.10
	GitPython                  3.1.30
	google-api-core            2.11.0
	google-auth                2.16.2
	google-cloud-bigquery      3.7.0
	google-cloud-core          2.3.2
	google-crc32c              1.5.0
	google-resumable-media     2.4.1
	googleapis-common-protos   1.58.0
	graphviz                   0.20.1
	grpcio                     1.49.1
	grpcio-status              1.51.3
	gunicorn                   20.1.0
	h11                        0.14.0
	hatch                      1.6.3
	hatchling                  1.13.0
	html5lib                   1.1
	httpcore                   0.16.3
	httpx                      0.23.3
	hyperlink                  21.0.0
	hyperopt                   0.2.7
	identify                   2.5.18
	idna                       3.4
	importlib-metadata         5.2.0
	indexed                    1.3.0
	iniconfig                  1.1.1
	ipykernel                  6.19.4
	ipytest                    0.13.1
	ipython                    8.8.0
	ipython-genutils           0.2.0
	isoduration                20.11.0
	itsdangerous               2.1.2
	jaraco.classes             3.2.3
	jedi                       0.18.2
	Jinja2                     3.1.2
	jinja2-ansible-filters     1.3.2
	jinja2-time                0.2.0
	jmespath                   1.0.1
	joblib                     1.2.0
	json5                      0.9.11
	jsonpointer                2.3
	jsonschema                 4.17.3
	jupyter_ai                 0.7.3
	jupyter_ai_magics          0.7.3
	jupyter_client             7.4.8
	jupyter_core               5.1.2
	jupyter-events             0.5.0
	jupyter-lsp                2.0.1
	jupyter-manip              0.1.0                           /Users/matt/Dropbox/work/pylib/jupyter_manip
	jupyter_server             2.5.0
	jupyter_server_fileid      0.6.0
	jupyter_server_terminals   0.4.3
	jupyter_server_ydoc        0.8.0
	jupyter-ydoc               0.2.4
	jupyterlab                 4.0.2
	jupyterlab-flake8          0.7.1
	jupyterlab-pygments        0.2.2
	jupyterlab_server          2.19.0
	jupytext                   1.14.6
	keyring                    23.13.1
	kiwisolver                 1.4.4
	langchain                  0.0.159
	langcodes                  3.3.0
	llvmlite                   0.39.1
	lxml                       4.9.2
	lz4                        4.0.2
	macholib                   1.16.2
	Mako                       1.2.4
	Markdown                   3.1.1
	markdown-it-py             2.1.0
	MarkupSafe                 2.1.2
	marshmallow                3.19.0
	marshmallow-enum           1.5.1
	matplotlib                 3.6.2
	matplotlib-inline          0.1.6
	mccabe                     0.7.0
	mdit-py-plugins            0.3.3
	mdurl                      0.1.2
	MealMaker                  0.1                             /Users/matt/Dropbox/work/pylib/MealMaker
	miniful                    0.0.6
	mistune                    2.0.4
	mlflow                     2.1.1
	modin                      0+untagged.2282.g56f15ac
	more-itertools             9.0.0
	msgpack                    1.0.5
	multidict                  6.0.4
	multitasking               0.0.11
	murmurhash                 1.0.9
	mutagen                    1.46.0
	mypy-extensions            1.0.0
	nbclassic                  0.4.8
	nbclient                   0.7.2
	nbconvert                  7.2.7
	nbformat                   5.7.1
	nest-asyncio               1.5.6
	networkx                   2.8.8
	nodeenv                    1.7.0
	notebook                   6.5.2
	notebook_shim              0.2.2
	numba                      0.56.4
	numexpr                    2.8.4
	numpy                      1.23.5
	oauthlib                   3.2.2
	openai                     0.27.7
	openapi-schema-pydantic    1.2.4
	openpyxl                   3.0.10
	oscrypto                   1.3.0
	packaging                  23.1
	pandas                     2.0.2
	pandoc-minted              0.2.0
	pandocfilters              1.5.0
	panflute                   2.3.0
	parso                      0.8.3
	pathspec                   0.11.0
	pathy                      0.10.1
	patsy                      0.5.3
	pefile                     2022.5.30
	pelican                    4.2.0
	pexpect                    4.8.0
	pickleshare                0.7.5
	Pillow                     9.4.0
	pip                        23.1.2
	platformdirs               2.6.2
	plotly                     5.11.0
	pluggy                     1.0.0
	plumbum                    1.8.1
	polars                     0.17.13
	ponder                     0.0.3
	pre-commit                 3.0.4
	preshed                    3.0.8
	prometheus-client          0.15.0
	prompt-toolkit             3.0.36
	proto-plus                 1.22.2
	protobuf                   4.21.12
	psutil                     5.9.4
	ptyprocess                 0.7.0
	pure-eval                  0.2.2
	py4j                       0.10.9.7
	pyarrow                    12.0.0
	pyasn1                     0.4.8
	pyasn1-modules             0.2.8
	pycodestyle                2.10.0
	pycparser                  2.21
	pycryptodomex              3.17
	pydantic                   1.10.7
	pydotplus                  2.0.2
	pyflakes                   3.0.1
	pyFUME                     0.2.25
	Pygments                   2.15.1
	PyInstaller                3.4
	PyJWT                      2.6.0
	PyMuPDF                    1.21.1
	pyOpenSSL                  23.0.0
	pyparsing                  3.0.9
	pyperclip                  1.8.2
	pyproject_hooks            1.0.0
	pyrsistent                 0.19.3
	PySide6                    6.4.2
	PySide6-Addons             6.4.2
	PySide6-Essentials         6.4.2
	pytest                     7.2.0
	pytest_check_links         0.8.0
	pytest-console-scripts     1.3.1
	pytest-cov                 4.0.0
	pytest-jupyter             0.6.2
	pytest-timeout             2.1.0
	python-dateutil            2.8.2
	python-json-logger         2.0.4
	pytz                       2023.3
	PyYAML                     6.0
	pyyaml-include             1.3
	pyzmq                      24.0.1
	querystring-parser         1.2.4
	questionary                1.10.0
	ray                        2.4.0
	regex                      2023.5.5
	requests                   2.28.1
	requests-cache             0.9.8
	rfc3339-validator          0.1.4
	rfc3986                    1.5.0
	rfc3986-validator          0.1.1
	rich                       13.3.1
	rpyc                       5.2.3
	rsa                        4.9
	rst2odp                    0.3.2                           /Users/matt/Dropbox/work/pylib/rst2odp
	rubber                     1.5.1.-1.6.0-alpha1-2-g0f20a84- /Users/matt/Dropbox/work/pylib/rubber3
	ruff                       0.0.246
	s3transfer                 0.6.0
	scikit-learn               1.2.0
	scikit-plot                0.3.7
	scipy                      1.10.0
	seaborn                    0.12.2
	Send2Trash                 1.8.0
	setuptools                 65.6.3
	shap                       0.41.0
	shellingham                1.5.0.post1
	shiboken6                  6.4.2
	simpful                    2.9.0
	six                        1.13.0
	sklearn                    0.0.post1
	slicer                     0.0.7
	smart-open                 6.3.0
	smartypants                2.0.1
	smmap                      5.0.0
	sniffio                    1.3.0
	snowflake-connector-python 3.0.0
	soupsieve                  2.3.2.post1
	spacy                      3.5.1
	spacy-legacy               3.0.12
	spacy-loggers              1.0.4
	sparklines                 0.4.2
	SQLAlchemy                 1.4.46
	sqlglot                    11.3.7
	sqlparse                   0.4.3
	srsly                      2.4.6
	stack-data                 0.6.2
	statsmodels                0.13.5
	svgslicer                  1.0.0                           /Users/matt/Dropbox/work/pylib/svgslicer
	svgwrite                   1.4.3
	tabulate                   0.9.0
	tenacity                   8.1.0
	terminado                  0.17.1
	thinc                      8.1.9
	threadpoolctl              3.1.0
	tiktoken                   0.4.0
	tinycss2                   1.2.1
	tokenize-rt                5.0.0
	toml                       0.10.2
	tomli                      2.0.1
	tomli_w                    1.0.0
	tomlkit                    0.11.6
	tornado                    6.2
	tqdm                       4.64.1
	traitlets                  5.8.0
	typer                      0.7.0
	typing_extensions          4.5.0
	typing-inspect             0.9.0
	typogrify                  2.0.7
	tzdata                     2023.3
	Unidecode                  1.1.1
	uri-template               1.2.0
	url-normalize              1.4.3
	urllib3                    1.26.13
	userpath                   1.8.0
	virtualenv                 20.19.0
	wasabi                     1.1.1
	wcwidth                    0.2.5
	webcolors                  1.12
	webencodings               0.5.1
	websocket-client           1.4.2
	websockets                 11.0.2
	Werkzeug                   2.2.2
	wheel                      0.38.4
	wxPython                   4.2.0
	xgboost                    1.7.2
	xlrd                       2.0.1
	xyzservices                2023.2.0
	y-py                       0.5.5
	yarl                       1.8.2
	yellowbrick                1.5
	yfinance                   0.2.6
	youtube-dl                 2021.12.17
	ypy-websocket              0.8.2
	zipp                       3.11.0

</pre>
</details>

**Comment by JasonWeill** [comment]
Are you using real-time collaboration (RTC) mode? If so, there's a known bug about scrolling to top: #14106. If not, this might be a separate issue.

**Comment by mattharrison** [comment]
I'm working solo. So new bug?

**Comment by hectormz** [comment]
I'm getting this behavior too

**Comment by michaelaye** [comment]
it's scrolling to the end of the 2nd cell of the split for me, while it should stay, IMHO, at the end of the 1st new cell and before the new 2nd , exactly where one did the split.

**Comment by andrii-i** [comment]
@michaelaye when cell is split, the view port is moved / scrolls to the end of the 2nd cell of the split but this different from "scrolls to the top" bug discussed here where viewport just scrolls to the top of the notebook. Note video above by @krassowski https://github.com/jupyterlab/jupyterlab/issues/14728#issuecomment-1749106424

**Comment by andrii-i** [comment]
Hi everyone. I'm having a hard time reproducing this unexpected scrolling when splitting a cell. Could you please see if you can still reproduce this behavior in latest version of the JupyterLab?
